metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from collections import OrderedDict
+from typing import Dict, Iterable, List
+from dataclasses import dataclass, field
+
+from datahub.configuration.source_common import PlatformSourceConfigBase
+from datahub.emitter.mce_builder import (
+    make_data_platform_urn,
+    make_dataset_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.aws.aws_common import AwsSourceConfig
+from datahub.metadata.schema_classes import (
+    ChangeTypeClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    BooleanType,
+    BytesType,
+    KeyValueSchema,
+    NumberType,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringType,
+)
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PLATFORM = "dynamodb"
+
+_attribute_type_mapping = {
+    "S": StringType,
+    "N": NumberType,
+    "B": BytesType,
+    "BOOL": BooleanType,
```

Comment:
According to this there is support for bool and more types as well:
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes

Reply:
yeah, it's confusing as they have inconsistent documentations.
here's another one: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_AttributeDefinition.html
we need the abbreviation forms like `S` for `String.

Reply:
This one gives abbreviations for the rest: http://man.hubwiz.com/docset/Boto3.docset/Contents/Resources/Documents/reference/services/dynamodbstreams.html#DynamoDBStreams.Client.get_records